### DISCUSSION TITLE
initiateOnDeviceConversionMeasurement()

### DISCUSSION THREAD

**Discussion by braxtonchristensen** [invertase/react-native-firebase#6264] [Q&A]
Is there any plans in the near future to add support for `initiateOnDeviceConversionMeasurement()` for iOS? Has anyone enabled this in their RN application, what did that process look like?

https://support.google.com/google-ads/answer/12119136

**Comment by mikehardy** [comment]
Looks like it will block on upgrade here to support iOS SDK v9, following that I think implementation path is to add the plugin pod with a podfile variable (my guess, similar to the with-or-without IDFA variable) then expose the API up from native to JavaScript

The hardest part of this one will undoubtedly be testing, testing cross product integration like this is difficult and we are not set up to do this well unfortunately as I'm unaware of programmatic apis we can use to prove correctness. If I generate a downloadable set of patch-package patches for the feature, would you be able to verify they are working?

**Reply by braxtonchristensen** [reply]
This sounds amazing! I think we are interested in testing that the feature is working correctly for this package. What version of `@react-native-firebase` would we be applying the patches to? I want to make sure that our application has the correct version of `@react-native-firebase/*` so we can be ready to use the patch-package patches whenever they are ready. Let me know if there is anything you need from me. ❤️

**Reply by ninaliu2021** [reply]
Hi Mike 👋😊. I'm Nina and I work with Braxton above. I wanted to follow up and see if you've decided to apply this patch, and if you are applying it to iOS SDK v9? 

We need this to enable attribution in our ad spend, and would love to support the testing efforts with our product. We can update you on our findings and make sure it's working properly. Thanks so much!

**Reply by mikehardy** [reply]
Hi there @ninaliu2021 I would like to support it, I'm still in the process of updating the library here to support firebase-ios-sdk v9+ so that we even have access to the APIs. I will support it after that is done. Hoping for this week - I'm hung up a little bit on https://github.com/firebase/firebase-ios-sdk/issues/9849 but we are close to resolution there, and I need https://github.com/firebase/firebase-ios-sdk/pull/9862 released, which should be in firebase-ios-sdk 9.2.0 (probably this week?)

Regardless of firebase-ios-sdk release status, once I have resolution on 9849 I can start working on this even if it's unreleased, and as I post PRs part of our automated PR process generates patch-package compatible patches, so testing can begin.

This is a bit less related but please note one of the biggest changes in firebase-ios-sdk v9 is requiring the specification of linkage in cocoapods Podfile via `use_frameworks!` and this has known problems with all react-native versions if you use Flipper or Hermes, and it has a known issue I'm tracking down with facebook via participation in release testers group where it does not work with upcoming react-native v0.69 (currently on rc6) at all. So for you to adopt the patches as they come out you'll need to be on react-native 0.68 or lower and you'll need to disable hermes and flipper at least until we determine some workarounds

**Reply by mikehardy** [reply]
That gives me an idea actually: you might try adding either `use_frameworks!` or `use_frameworks! :linkage => :static` to your Podfile now along with `$RNFirebaseStaticFrameworks = true` now to make sure things continue building for you. That *should* work even on firebase-ios-sdk v8 unless you have some other 3rd party incompatibility. Best to start checking that now 

https://github.com/mikehardy/rnfbdemo/blob/51ef9f125d67b599854de81764b9451514a5608e/make-demo.sh#L274

https://rnfirebase.io/#allow-ios-static-frameworks

**Reply by braxtonchristensen** [reply]
That is terrific news! I have added those lines to our `Podfile` and everything seems to be working great! Thank you very much, Mike. We appreciate you. ❤️

**Reply by mikehardy** [reply]
firebase-ios-sdk v9 just shipped with v15 here - #6274 - so this is unblocked. I logged it here for tracking #6321

**Reply by mikehardy** [reply]
PR #6325 is up for testing now - and as I type this it looks like our continuous integration workflows have already generated a [patch-package set](https://github.com/invertase/react-native-firebase/actions/runs/2538945242) for local testing integration. Would love feedback on the PR itself if you have testing results :pray:
